ape_safe/factory.py

```diff
@@ -0,0 +1,115 @@
+import secrets
+from functools import cache
+from typing import TYPE_CHECKING, Iterable, Optional, Union
+
+from ape.types import AddressType
+from ape.utils import ZERO_ADDRESS, ManagerAccessMixin
+from packaging.version import Version
+
+from ape_safe.packages import SAFE_PACKAGE_BY_VERSION, PackageType, get_factory, get_singleton
+
+if TYPE_CHECKING:
+    from ape.api import AccountAPI
+    from ape.contracts import ContractInstance
+
+
+class Factory(ManagerAccessMixin):
```

Comment:
`Factory` is a bit generic for this class's name... Maybe `SafeFactory` would be better